apps/aecore/test/aec_peers_pool_tests.erl

```diff
@@ -0,0 +1,1722 @@
+%%%=============================================================================
+%%% @copyright (C) 2017, Aeternity Anstalt
+%%% @doc
+%%%   Unit tests for the aec_peers module
+%%% @end
+%%%=============================================================================
+-module(aec_peers_pool_tests).
+
+-ifdef(TEST).
+
+%=== INCLUDES ==================================================================
+
+-include_lib("eunit/include/eunit.hrl").
+
+%=== TEST CASES ================================================================
+
+%% Tests pool behavior when empty.
+empty_pool_test() ->
+  P = aec_peers_pool:new([]),
+  A = random_peer_id(),
```

Comment:
For static unit tests random data should not be used since it is pointless and makes it harder to debug (it can also some unintentional results, such as by chance generating correct data when it shouldn't). It is better to write a QuickCheck specification in that case.

Reply:
In this case where only one id/address is used it would be fine, but lots of tests use 10th or 100th of id/addresses and having them static would be a pain.

Reply:
`lists:seq(1,100)`

Reply:
This give you 1-- integer, no 100 binaries with realistic size or ip addresses with realistic values... using random one make it simpler for that...

Reply:
Obviously you would have to use the integers for something, I'm just saying it's easy to enumerate a fixed set of something. My original argument still stands.

Reply:
I made sure it was possible to make the tests 100% predictable even though using a random generator. You can specify the seed and disable strong random offset. I used it multiple time to investigate bugs while writing the tests. I could ALWAYS set the seeds but I don't feel it give us anything until some tests randomly fail...

Reply:
Then why not write QuickCheck tests? Currently, it's the "worst of both worlds." We don't have predictable static unit tests that can be re-run with confidence, and we don't get the benefits of smart generation and shrinking from QC.

Reply:
Because I don't have a license for QC and still need to be sure the code works?
And because I feel it is simpler for simple tests to just generate ids/addresses...